baremetal/src/include/ebbrt/Perf.h

```diff
@@ -0,0 +1,179 @@
+//          Copyright Boston University SESA Group 2013 - 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+//
+#ifndef BAREMETAL_SRC_INCLUDE_EBBRT_PERF_H_
+#define BAREMETAL_SRC_INCLUDE_EBBRT_PERF_H_
+#include <ebbrt/CpuAsm.h>
+
+namespace ebbrt {
+namespace perf {
+#define PERF_CPUID_LEAF 0x0A
+#define FIXED_CTL_RING_LVL_NONE 0x0
+#define FIXED_CTL_RING_LVL_ALL 0x3
+#define IA32_FIXED_CTR_CTRL_MSR 0x38D
+#define IA32_PERF_GLOBAL_CTRL_MSR 0x38F
+#define IA32_PERF_GLOBAL_STATUS_MSR 0x38E
+#define IA32_PERF_GLOBAL_OVF_CTRL_MSR 0x390
+#define IA32_FXD_PMC(X) (0x309 + (X))
+#define IA32_PERFEVTSEL_MSR(X) (0x186 + (X))
+#define IA32_PMC(X) (0x0C1 + (X))
+#define FIXED_EVT_OFFSET(X) (0x20 + (X))
+#define NIL_EVT_OFFSET 0xFF
+// cycles
+#define PERFEVTSEL_EVT_CYCLES 0x3C
+#define PERFEVTSEL_UMASK_CYCLES 0x00
+// reference cycles
+#define PERFEVTSEL_EVT_CYCLES_REF 0x3C
+#define PERFEVTSEL_UMASK_CYCLES_REF 0x01
+// instructions
+#define PERFEVTSEL_EVT_INSTRUCTIONS 0xC0
+#define PERFEVTSEL_UMASK_INSTRUCTIONS 0x00
+// last-level cache references
+#define PERFEVTSEL_EVT_LLC_REF 0x2E
+#define PERFEVTSEL_UMASK_LLC_REF 0x4F
+// last-level cache misses
+#define PERFEVTSEL_EVT_LLC_MISSES 0x2E
+#define PERFEVTSEL_UMASK_LLC_MISSES 0x41
+// branch instructions
+#define PERFEVTSEL_EVT_BRANCH_INSTRUCTIONS 0xC4
+#define PERFEVTSEL_UMASK_BRANCH_INSTRUCTIONS 0x00
+// branch misses
+#define PERFEVTSEL_EVT_BRANCH_MISSES 0xC5
+#define PERFEVTSEL_UMASK_BRANCH_MISSES 0x00
+// (Intel Xeon) TLB miss all levels
+#define PERFEVTSEL_EVT_TLB_STORE_MISSES 0x49
+#define PERFEVTSEL_UMASK_TLB_STORE_MISSES 0x0E
+
+#define PERFEVTSEL_EVT_TLB_LOAD_MISSES 0x08
+#define PERFEVTSEL_UMASK_TLB_LOAD_MISSES 0x01
```

Comment:
I'd rather see all these typed correctly like so:
`const constexpr uint8_t kUmaskTlbLoadMisses = 0x01;`

Also the function macros can be made into constexpr functions.